### PR TITLE
Use app-internal organization for TC

### PR DESCRIPTION
**Pull request by enocom**
- Split TC into proxy and auth packages
- Rename auth.New (was accessLogger.New) to auth.NewAccessLogger
- Move `startOutgoingProxy` function to method on traffic controller
- Extract shared gRPC connector and pool into plumbing since the rlp
also uses it.
- Remove useless negative test in log_access_authorizer_test that
checked for extraneous goroutines.
- Update bosh package specs for rlp and loggregator_trafficcontroller

### CODE REVIEW

File: src/trafficcontroller/app/traffic_controller.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+		dopplerHandler = accessMiddleware(dopplerHandler)
+	}
+	t.startOutgoingProxy(fmt.Sprintf(":%d", t.conf.OutgoingDropsondePort), dopplerHandler)
+
+	killChan := signalmanager.RegisterKillSignalChannel()
+	dumpChan := signalmanager.RegisterGoRoutineDumpSignalChannel()
+
+	// We start the profiler last so that we can definitively claim that we're ready for
+	// connections by the time we're listening on the PPROFPort.
+	p := profiler.New(t.conf.PPROFPort)
```
Comment (jasonkeene) [inline src/trafficcontroller/app/traffic_controller.go]: On other components we put the pprof binding in the main.

https://github.com/cloudfoundry-incubator/scalable-syslog/blob/0fa3b79fa50fc1b12658e8ec0f9e19829205a512/adapter/main.go#L49
https://github.com/cloudfoundry-incubator/scalable-syslog/blob/0fa3b79fa50fc1b12658e8ec0f9e19829205a512/scheduler/main.go#L53

I'm not sure all of this belongs in the app.

**Comment by enocom** [comment]
@jasonkeene The profiler may not belong in the app, but this PR here is solely about the organization of files. I'd prefer to fix the TC app and improve its factoring in another PR, if that's alright with you.

**Comment by jasonkeene** [comment]
Sure thing I will merge then.

**Pull request merged**